components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/util/RandomAttrGeneratorFactoryImpl.java

```diff
@@ -93,16 +93,16 @@ public void validateRandomAttrGenerator(JSONObject attributeConfig, Attribute.Ty
             }
             switch (type) {
                 case CUSTOM_DATA_BASED:
-                    new CustomBasedAttrGenerator().validateAttributeConfiguration(attributeConfig);
+                    new CustomBasedAttrGenerator().validateAttributeConfiguration(attrType, attributeConfig);
                     break;
                 case PRIMITIVE_BASED:
-                    new PrimitiveBasedAttrGenerator(attrType).validateAttributeConfiguration(attributeConfig);
+                    new PrimitiveBasedAttrGenerator().validateAttributeConfiguration(attrType, attributeConfig);
                     break;
                 case PROPERTY_BASED:
-                    new PropertyBasedAttrGenerator().validateAttributeConfiguration(attributeConfig);
+                    new PropertyBasedAttrGenerator().validateAttributeConfiguration(attrType, attributeConfig);
                     break;
                 case REGEX_BASED:
-                    new RegexBasedAttrGenerator().validateAttributeConfiguration(attributeConfig);
+                    new RegexBasedAttrGenerator().validateAttributeConfiguration(attrType, attributeConfig);
```

Comment:
Different types of attribute generators are created. Validation logic varies based on the random generation type.